### PR TITLE
Added ability to set 'Description' field when calling New-AzADGroup (Issue #9004)

### PR DESCRIPTION
**Pull request by lwajswaj**
## Description

This PR adds 'Description' parameter to New-AzAdGroup cmdlet (Issue #9004) as well as adds it as output when calling Get-AzAdGroup. Also, fixed value displayed as MailNickName.

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Resources/Resources/ActiveDirectory/Models/ActiveDirectoryClientExtensions.cs [hunk 1]
```diff
@@ -60,7 +60,8 @@ public static PSADObject ToPSADObject(this AADObject obj)
                     Id = obj.ObjectId,
                     Type = obj.ObjectType,
                     SecurityEnabled = obj.SecurityEnabled,
-                    MailNickname = obj.Mail
+                    MailNickname = (obj.Mail != null ? obj.Mail : obj.AdditionalProperties["mailNickname"].ToString()),
```
Comment (MiYanni) [inline src/Resources/Resources/ActiveDirectory/Models/ActiveDirectoryClientExtensions.cs]: Can simply be:
```suggestion
                    MailNickname = obj.Mail ?? obj.AdditionalProperties["mailNickname"].ToString(),
```

File: src/Resources/Resources/ActiveDirectory/Models/ActiveDirectoryClientExtensions.cs [hunk 2]
```diff
@@ -60,7 +60,8 @@ public static PSADObject ToPSADObject(this AADObject obj)
...
+                    Description = (obj.AdditionalProperties.ContainsKey("description") && obj.AdditionalProperties["description"] != null ? obj.AdditionalProperties["description"].ToString() : null)
```
Comment (MiYanni) [inline src/Resources/Resources/ActiveDirectory/Models/ActiveDirectoryClientExtensions.cs]: Can simply be:
```suggestion
                    Description = obj.AdditionalProperties.ContainsKey("description") ? obj.AdditionalProperties["description"]?.ToString() : null
```

File: src/Resources/Resources/ActiveDirectory/Models/ActiveDirectoryClientExtensions.cs [hunk 3]
```diff
@@ -110,7 +111,8 @@ public static PSADGroup ToPSADGroup(this ADGroup group)
                 DisplayName = group.DisplayName,
                 Id = group.ObjectId,
                 SecurityEnabled = group.SecurityEnabled,
-                MailNickname = group.Mail
+                MailNickname = (group.Mail != null ? group.Mail : group.AdditionalProperties["mailNickname"].ToString()),
```
Comment (MiYanni) [inline src/Resources/Resources/ActiveDirectory/Models/ActiveDirectoryClientExtensions.cs]: Can simply be:
```suggestion
                MailNickname = group.Mail ?? group.AdditionalProperties["mailNickname"].ToString(),
```

File: src/Resources/Resources/ActiveDirectory/Models/ActiveDirectoryClientExtensions.cs [hunk 4]
```diff
@@ -110,7 +111,8 @@ public static PSADGroup ToPSADGroup(this ADGroup group)
...
+                Description = (group.AdditionalProperties.ContainsKey("description") && group.AdditionalProperties["description"] != null ? group.AdditionalProperties["description"].ToString() : null)
```
Comment (MiYanni) [inline src/Resources/Resources/ActiveDirectory/Models/ActiveDirectoryClientExtensions.cs]: Can simply be:
```suggestion
                Description = group.AdditionalProperties.ContainsKey("description") ? group.AdditionalProperties["description"]?.ToString() : null
```

**Comment by lwajswaj** [comment]
Thanks @MiYanni for those one-liners, just learnt about ?.ToString()

**Comment by MiYanni** [comment]
@darshanhs90 Can you take a look at this PR?

**Pull request merged**